src/components/FormStep.js

```diff
@@ -304,59 +309,70 @@ const FormStep = ({
-                  <Button
-                    type="submit"
-                    variant="primary"
-                    name="next"
-                    disabled={!canSubmit}
-                  ><Literal name="nextText"/></Button>
-                </ToolbarList>
-              </Toolbar>
-            </LiteralsProvider>
-            {form.loginRequired ? <LogoutButton onLogout={onLogout}/> : null}
-          </form>
-        ) : null
-      }
-    </Card>
+    <>
+      <Card title={submissionStep.name}>
+        { loading ? <Loader modifiers={['centered']} /> : null }
+
+        {
+          (!loading && configuration) ? (
+            <form onSubmit={onReactSubmit}>
+              <FormIOWrapper
+                ref={formRef}
+                form={configuration}
+                // Filter blank values so FormIO does not run validation on them
+                submission={{data: filterBlankValues(data)}}
+                onChange={onFormIOChange}
+                onSubmit={onFormIOSubmit}
+                options={{
+                  noAlerts: true,
+                  baseUrl: config.baseUrl,
+                  language: formioTranslations.language,
+                  i18n: formioTranslations.i18n,
+                  hooks,
+                  intl
+                }}
+              />
+              { DEBUG ? <FormStepDebug data={data} /> : null }
+              <LiteralsProvider literals={formStep.literals}>
+                <Toolbar modifiers={['mobile-reverse-order', 'bottom']}>
+                  <ToolbarList>
+                    <Button
+                      variant="anchor"
+                      component="a"
+                      onClick={onPrevPage}
+                    ><Literal name="previousText"/></Button>
+                  </ToolbarList>
+                  <ToolbarList>
+                    {/* Hiding the Save button until the functionality is implemented */}
+                    <Button
+                      type="button"
+                      variant="secondary"
+                      name="save"
+                      onClick={onFormSave}
+                    ><Literal name="saveText"/></Button>
+                    <Button
+                      type="submit"
+                      variant="primary"
+                      name="next"
+                      disabled={!canSubmit}
+                    ><Literal name="nextText"/></Button>
+                  </ToolbarList>
+                </Toolbar>
+              </LiteralsProvider>
+              {form.loginRequired ? <LogoutButton onLogout={onLogout}/> : null}
+            </form>
+          ) : null
+        }
+      </Card>
+      <FormStepSaveModal
+        isOpen={shouldFormStepSaveModalBeOpen}
+        closeModal={closeFormStepSaveModal}
+        stepData={{...formData.current}}
+        saveStepDataUrl={submissionStep.url}
+        suspendFormUrl={`${submission.url}/_suspend`}
+        destroySessionUrl={`${config.baseUrl}authentication/session`}
```

Comment:
fetch the `config` from the context in the downstream component rather than passing it as prop, this makes it easier to re-arrange the components and loosens coupling.